vm/devices/tpm/src/ak_cert.rs

```diff
@@ -19,6 +19,20 @@ pub enum TpmAkCertType {
     HwAttested(Box<dyn RequestAkCert>),
 }
 
+impl TpmAkCertType {
+    /// Get the `RequestAkCert` from the enum
+    ///
+    /// The function must only be called if the enum is [`TpmAkCertType::HwAttested`]
+    /// or [`TpmAkCertType::Trusted`].
+    pub fn get_ak_cert_helper(&mut self) -> &mut Box<dyn RequestAkCert> {
```

Comment:
I don't think this needs to be `&mut`. A `Option<&Box<dyn RequestAkCert>>` seems sufficient here.

Reply:
on a semi-related note - you may want to swap this over to using `Arc<dyn>` as well, so you can avoid the need for `cloned_box`